src/screens/ProductDetails/ProductDetails.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import Screen from '../../styledComponents/Screen'
+import Button from '../../styledComponents/Button'
+import products from '../../data/products'
+import categories from '../../data/categories'
+import {connect} from 'react-redux'
+import { Link } from 'react-router-dom'
+import {addToCart} from '../../screens/ShoppingCart/actionCreators'
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    addToCart: (product,amount) => dispatch(addToCart(product,amount))
+  }
+}
+
+const ProductDetails = (props) => {
+
+  const id = parseInt(props.match.params.id, 10)
+  const product = products.find(product => product.id === id)
+  const category = categories.find(category => category.id === product.category).name
```

Comment:
Very very small issue, but maybe for a sake of symmetry with const product, category should remain the whole item, not .name?